Nested facet - add custom nested type and change presenter logic

lauraghiorghisor-tw opened the pull request:
Changes:
- introduce custom "nested" type to replace `nested_facet` flag
- revert presenter logic so that we send a single cohesive "nested" load to Publishing API.

### Admin area changes
- New type value
> ![Screenshot 2025-03-13 at 10 37 18](https://github.com/user-attachments/assets/68b46ff9-a442-4919-acc9-d554c9ae51b4)

- Existing Grounds section facet UI
> ![Screenshot 2025-03-13 at 10 32 24](https://github.com/user-attachments/assets/b1d2bd66-ffba-4bdf-9e42-7d297a875e89)

- Existing grounds section facet generated schema in admin area (no change other than type)
> ![Screenshot 2025-03-12 at 13 18 16](https://github.com/user-attachments/assets/407be773-79df-4a35-a428-869052dde76e)

- New facet added with nested multiple type - diff in admin area (no change other than type)
> ![Screenshot 2025-03-12 at 13 19 07](https://github.com/user-attachments/assets/7f09a8c3-b0e8-4d45-b717-7f42d8c42e87)

- New facet added with nested single type - diff in admin area (no change other than type)
> ![Screenshot 2025-03-13 at 10 36 28](https://github.com/user-attachments/assets/d3b03ba3-3a49-47d3-93f5-dd2715edb723)



[Trello](https://trello.com/c/EeZOA3iW/3534-spike-replicating-taxons-approach-for-nested-facets)

lauraghiorghisor-tw committed:
Replace `nested_facet` flag with `nested` type

Introduce a clearer distinction between the text and nested facet type.

lauraghiorghisor-tw committed:
Revert presenter changes for  nested facet

We're changing implementation to have a single facet in Finder Frontend, rather than two.

We are nonetheless keeping the `main_facet_label` and `main_facet_value` references in the nested structure.

minhngocd commented on app/models/facet.rb:
```diff
@@ -115,7 +112,7 @@ def facet_specialist_publisher_properties(type, validations)
 
     def facet_specialist_publisher_properties_select(type)
       case type
-      when "enum_text_multiple"
```
nested facets could be used for both single select and multiselect fields, no?

lauraghiorghisor-tw replied:
Ah, so add... two options here? Nested single and nested multiple?

lauraghiorghisor-tw replied:
Pushed as a new commit (last commit).

lauraghiorghisor-tw replied:
And added new screenshots.

lauraghiorghisor-tw committed:
Split nested options into multiple and single options

The nested type is not enough to cover the previous options we had, where the "nested_facet" option dictated whether the facet was nested, and the type itself gave us the 'specialist_publisher_property" of multiple/one.
Introducing a single type to cover both aspects must thus be split in two:
- nested with a single selectable option (value is set to `nested_enum_text_single`)
- nested with multiple selectable options (value is set to `nested_enum_text_multiple`).

minhngocd reviewed (approved):
LGTM - out of scope for this PR but maybe we need to revisit view logic splitting from models at some point as i had a little bit of trouble following the "type" changes.

dependabot[bot] committed:
Bump gds-api-adapters from 98.3.0 to 98.4.0

Bumps [gds-api-adapters](https://github.com/alphagov/gds-api-adapters) from 98.3.0 to 98.4.0.
- [Changelog](https://github.com/alphagov/gds-api-adapters/blob/main/CHANGELOG.md)
- [Commits](https://github.com/alphagov/gds-api-adapters/compare/v98.3.0...v98.4.0)

---
updated-dependencies:
- dependency-name: gds-api-adapters
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
